### PR TITLE
MULE-15667: Grizzly should reset requestInput stream in NTLM type3 message. #7

### PR DESCRIPTION
**Pull request by fsgonz**

### CODE REVIEW

File: src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java [hunk 1]
```diff
@@ -48,10 +48,17 @@ public InputStream getInputStream() {
...
     @Override
     public Body createBody() throws IOException {
-        if (inputStream.markSupported()) {
-          inputStream.mark(0);
+      if (inputStream.markSupported()) {
+        try {
           inputStream.reset();
+        } catch (IOException ioe) {
+          if (logger.isDebugEnabled()) {
+            logger.debug("Unable to reset the input stream: %s", ioe.getMessage());
```
Comment (elrodro83) [inline src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java]: does `%s` work here? or have to use `{}`?

**Review by elrodro83 — CHANGES_REQUESTED** [review]
Cahnge `PayloadGenFactory` line 277 to check for `isMarkSupprted` before attempting reset.

File: src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java (hunk 1 above)
Comment (fsgonz) [reply src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java]: No, you were right. Braces must be used.

File: src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java [hunk 2]
```diff
@@ -46,12 +46,22 @@ public InputStream getInputStream() {
         return inputStream;
     }
 
+    public static void main(String[] args) {
```
Comment (elrodro83) [inline src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java]: remove

File: src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java [hunk 3]
```diff
@@ -46,12 +46,22 @@ public InputStream getInputStream() {
...
           inputStream.reset();
+        } catch (IOException ioe) {
+          if (logger.isDebugEnabled()) {
+            logger.debug("Unable to reset the input stream: {}", ioe.getMessage());
+          }
         }
+
+        inputStream.mark(0);
+      }
         return new ISBody();
```
Comment (elrodro83) [inline src/main/java/com/ning/http/client/generators/InputStreamBodyGenerator.java]: format

**Review by elrodro83 — CHANGES_REQUESTED** [review]

**Review by elrodro83 — APPROVED** [review]

**Pull request merged**